signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWithAppleButton.kt

```diff
@@ -60,14 +59,14 @@ class SignInWithAppleButton @JvmOverloads constructor(
         (background as? GradientDrawable)?.cornerRadius = cornerRadius
 
         if (icon != -1) {
-            imageView.setImageResource(icon)
+            textView.setCompoundDrawablesRelativeWithIntrinsicBounds(icon, 0, 0, 0)
 
             if (iconSize != -1) {
-                imageView.layoutParams.width = iconSize
-                imageView.layoutParams.height = iconSize
+                textView.layoutParams.width = iconSize
+                textView.layoutParams.height = iconSize
```

Comment:
I like that you've updated an ImageView + TextView to be a TextView with a start drawable! I don't think this setting of layout parameters is required anymore; the icon size was applied to the ImageView because it contained the icon, but the icon is now sized according to its intrinsic bounds when set as a compound drawable on the TextView.